docs/posts/1brc/index.qmd

```diff
@@ -0,0 +1,73 @@
+---
+title: "1 billion row challenge with Ibis and DuckDB"
+author: ""
+date: "2024-01-40"
+categories:
+    - blog
+    - duckdb
+---
+
+## Overview
+
+https://github.com/gunnarmorling/1brc
+
+```{.bash}
+gh repo clone gunnarmorling/1brc
+```
+
+```{.bash}
+cd 1brc/src/main/python
+python create_measurements.py 1_000_000_000
+```
+
+```{python}
+import ibis
+
+ibis.set_backend("polars")
+
+ibis.options.interactive = True
+```
+
+```{python}
+if ibis.get_backend().name == "duckdb":
+    t = ibis.read_csv(
+        "1brc/data/measurements.txt",
+        delim=";",
+        header=False,
+        columns={"station": "VARCHAR", "temperature": "DOUBLE"},
+    )
+elif ibis.get_backend().name == "polars":
```

Comment:
Can you instead collect the kwargs into a dict and then have a single call to `ibis.read_csv`? It's pretty noisy with the repetition.